src/SourceLocator/SourceStubber/PhpStormStubsSourceStubber.php

```diff
@@ -107,37 +122,53 @@ public function generateClassStub(string $className) : ?StubData
 
     public function generateFunctionStub(string $functionName) : ?StubData
     {
-        if (! array_key_exists($functionName, PhpStormStubsMap::FUNCTIONS)) {
+        $lowercaseFunctionName = strtolower($functionName);
+
+        if (! array_key_exists($lowercaseFunctionName, self::$functionMap)) {
             return null;
         }
 
-        $filePath = PhpStormStubsMap::FUNCTIONS[$functionName];
+        $filePath = self::$functionMap[$lowercaseFunctionName];
 
-        if (! array_key_exists($functionName, $this->functionNodes)) {
+        if (! array_key_exists($lowercaseFunctionName, $this->functionNodes)) {
             $this->parseFile($filePath);
         }
 
-        return new StubData($this->createStub($this->functionNodes[$functionName]), $this->getExtensionFromFilePath($filePath));
+        return new StubData($this->createStub($this->functionNodes[$lowercaseFunctionName]), $this->getExtensionFromFilePath($filePath));
     }
 
     public function generateConstantStub(string $constantName) : ?StubData
     {
-        // https://github.com/JetBrains/phpstorm-stubs/pull/591
-        if (in_array($constantName, ['TRUE', 'FALSE', 'NULL'], true)) {
-            $constantName = strtolower($constantName);
+        $lowercaseConstantName = strtolower($constantName);
+
+        if (! array_key_exists($lowercaseConstantName, self::$constantMap)) {
+            return null;
         }
 
-        if (! array_key_exists($constantName, PhpStormStubsMap::CONSTANTS)) {
+        if (array_key_exists($lowercaseConstantName, $this->constantNodes)
+            && $this->constantNodes[$lowercaseConstantName] === null
+        ) {
             return null;
         }
 
-        $filePath = PhpStormStubsMap::CONSTANTS[$constantName];
+        $filePath     = self::$constantMap[$lowercaseConstantName];
+        $constantNode = $this->constantNodes[$constantName] ?? $this->constantNodes[$lowercaseConstantName] ?? null;
 
-        if (! array_key_exists($constantName, $this->constantNodes)) {
+        if ($constantNode === null) {
             $this->parseFile($filePath);
+
+            if (! array_key_exists($constantName, $this->constantNodes)
+                && ! array_key_exists($lowercaseConstantName, $this->constantNodes)
```

Comment:
would it be better to index `$this->constantNodes` by `lowercase-string` instead of performing a double lookup here?

Reply:
Constants are not case insensitive by default. They are case insensitive only if you use third parameter of `define`.

That's why I try both versions.

Reply:
makes sense. wondering if we can make this more explicit by stashing things into a `private` method, or by adding at least a comment...